var/spack/repos/builtin/packages/mvapich/package.py

```diff
@@ -207,48 +208,15 @@ def flag_handler(self, name, flags):
 
         return (flags, None, None)
 
-    def setup_build_environment(self, env):
-        # mvapich2 configure fails when F90 and F90FLAGS are set
-        env.unset("F90")
-        env.unset("F90FLAGS")
-
     def setup_run_environment(self, env):
         env.set("MPI_ROOT", self.prefix)
-
         # Because MPI functions as a compiler, we need to treat it as one and
         # add its compiler paths to the run environment.
-        self.setup_compiler_environment(env)
+        self.setup_mpi_wrapper_variables(env)
 
     def setup_dependent_build_environment(self, env, dependent_spec):
-        self.setup_compiler_environment(env)
-
-        # use the Spack compiler wrappers under MPI
-        dependent_module = dependent_spec.package.module
-        env.set("MPICH_CC", dependent_module.spack_cc)
-        env.set("MPICH_CXX", dependent_module.spack_cxx)
-        env.set("MPICH_F77", dependent_module.spack_f77)
-        env.set("MPICH_F90", dependent_module.spack_fc)
-        env.set("MPICH_FC", dependent_module.spack_fc)
-
-    def setup_compiler_environment(self, env):
-        env.set("MPICC", join_path(self.prefix.bin, "mpicc"))
-        env.set("MPICXX", join_path(self.prefix.bin, "mpicxx"))
-        env.set("MPIF77", join_path(self.prefix.bin, "mpif77"))
-        env.set("MPIF90", join_path(self.prefix.bin, "mpif90"))
-
-    def setup_dependent_package(self, module, dependent_spec):
-        self.spec.mpicc = join_path(self.prefix.bin, "mpicc")
-        self.spec.mpicxx = join_path(self.prefix.bin, "mpicxx")
-        self.spec.mpifc = join_path(self.prefix.bin, "mpif90")
-        self.spec.mpif77 = join_path(self.prefix.bin, "mpif77")
-
-    @run_before("configure")
-    def die_without_fortran(self):
-        # Until we can pass variants such as +fortran through virtual
-        # dependencies depends_on('mpi'), require Fortran compiler to
-        # avoid delayed build errors in dependents.
-        if (self.compiler.f77 is None) or (self.compiler.fc is None):
-            raise InstallError("Mvapich2 requires both C and Fortran compilers!")
+        self.setup_mpi_wrapper_variables(env)
+        super().setup_dependent_build_environment(env, dependent_spec)
```

Comment:
I think this breaks if the inheritance order ever changes, maybe better to explicitly call the parent classes that provide this method with `self`

```python
super(MpichEnvironmentModifications, self).setup_dependent_build_environment(...)
```